pkg/tsdb/loki/api.go

```diff
@@ -163,6 +163,14 @@ func (api *LokiAPI) DataQuery(ctx context.Context, query lokiQuery) (data.Frames
 	iter := jsoniter.Parse(jsoniter.ConfigDefault, resp.Body, 1024)
 	res := converter.ReadPrometheusStyleResult(iter, converter.Options{MatrixWideSeries: false, VectorWideSeries: false})
 
+	if res == nil {
```

Comment:
would it be better if `ReadPrometheusStyleResult` always returned a result instead of a pointer?

Reply:
yes, i agree. i created https://github.com/grafana/grafana/issues/52430 for that.